### PR TITLE
The solution of the 'payment service' problem

### PR DESCRIPTION
**Pull request by koryun23**

### CODE REVIEW

File: src/main/com/aca/homework/week8/payment/service/PaymentService.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.aca.homework.week8.payment.service;
+
+import java.util.Random;
+
+@Service
+public class PaymentService {
+    public static int sleep() {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/payment/service/PaymentService.java:7]: make this method private

File: src/main/com/aca/homework/week8/payment/service/Result.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+public class Result {
+    private final int waitingTime;
+    private final int paymentAmount;
+
+    public Result(int time, int pay) {
+        waitingTime = time;
+        paymentAmount = pay;
+    }
+
+    public static Result resultOf(int time, int pay) {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/payment/service/Result.java:12]: no need to create a static factory here

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

File: src/main/com/aca/homework/week8/payment/service/PaymentService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.aca.homework.week8.payment.service;
+
+import java.util.Random;
+
+@Service
+public class PaymentService {
+    private static int sleep() {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/payment/service/PaymentService.java:7]: please reorder methods. first should be public methods then private ones

File: src/test/com/aca/homework/week8/payment/service/PaymentServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.lang.annotation.Annotation;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PaymentServiceTest {
+    private PaymentService service;
+    private Result result;
```
Comment (gevorgyanarman) [inline src/test/com/aca/homework/week8/payment/service/PaymentServiceTest.java:12]: there is no need to keep Result as a global variable.

File: src/test/com/aca/homework/week8/payment/service/PaymentServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    @BeforeEach
+    public void setUp(){
+        service = new PaymentService();
+    }
+
+    @Test
+    public void testServiceAnnotation(){
+        Annotation[] annotations = service.getClass().getDeclaredAnnotations();
+        boolean foundServiceAnnotation = false;
+        for(Annotation annotation : annotations){
```
Comment (gevorgyanarman) [inline src/test/com/aca/homework/week8/payment/service/PaymentServiceTest.java:23]: please format the code

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

**Pull request merged**